app/controllers/foreman_tasks/api/tasks_controller.rb

```diff
@@ -141,6 +141,30 @@ def bulk_cancel
         }
       end
 
+      api :POST, '/tasks/bulk_stop', N_('Stop all stoppable tasks')
+      param :search, String, :desc => N_('Stop tasks matching search string')
+      param :task_ids, Array, :desc => N_('Stop specific tasks by ID')
+      def bulk_stop
+        if params[:search].nil? && params[:task_ids].nil?
+          raise BadRequest, _('Please provide at least one of search or task_ids parameters in the request')
+        end
+
+        filtered_scope = bulk_scope
+        to_stop = filtered_scope.where.not(state: :stopped)
+        already_stopped_length = filtered_scope.length - to_stop.length
+        to_stop.update_all(state: :stopped)
+
+        if params[:search]
+          notification = UINotifications::Tasks::TaskBulkStop.new(filtered_scope.first, to_stop.length, already_stopped_length)
+          notification.deliver!
+        end
+        render :json => {
+          total: filtered_scope.length,
```

Comment:
It would be nice if you could get the total count earlier in the code and then just reuse that, instead of counting the tasks in several places

Reply:
Changed, thanks